saleor/plugins/avatax/plugin.py

```diff
@@ -636,10 +636,14 @@ def calculate_order_shipping(
                     gross = Money(amount=net + tax, currency=currency)
                     net = Money(amount=net, currency=currency)
                 return TaxedMoney(net=net, gross=gross)
+
+        # Ignore typing checks because it is checked in _validate_order
+        price = order.shipping_method.channel_listings.get(  # type: ignore
```

Comment:
Are we sure that this channel always listing exists in the database at this point? I'm thinking about a race condition when it is deleted while this function is processing.

Reply:
The same is in [manager.py](https://github.com/saleor/saleor/blob/3.0/saleor/plugins/manager.py#L270), so maybe we should move validation there.